torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -493,6 +500,10 @@ class TORCH_API ProcessGroupNCCL : public Backend {
   // may indicate that there is some sort of collective desynchronization.
   uint64_t getSequenceNumberForGroup() override;
 
+  // Return the total number of splits the communicators held by this process
+  // group have performed.
+  uint64_t getCommSplitCounter() const;
```

Comment:
nit: it seems this API is for testing only. Do we need this API for other purpose?

Reply:
I can only think of testing and maybe observability (might be a useful counter to expose someday).  But for now just testing.  Is there another preferred way for exposing test-only counters?